Cargo.toml

```diff
@@ -9,18 +9,16 @@ version = "0.1.0"
 edition = "2018"
 
 [dependencies]
-cortex-m = { version = "0.5.4", features = [ "const-fn" ] }
-cortex-m-rt = "0.6.12"
-embedded-hal = "0.2.3"
-nrf52840-hal = "0.10.0"
+cortex-m = "0.7.3"
+embedded-hal = "0.2.6"
+nrf52840-hal = "0.13.0"
 
 [dev-dependencies]
-cortex-m-rt = "0.6.12"
-cortex-m-semihosting = "~0.3"
-panic-semihosting = "~0.5"
-nb = "~0.1"
-panic-rtt-target = { version = "0.1", features = ["cortex-m"] }
-rtt-target = { version = "0.2", features = ["cortex-m"] }
+cortex-m-rt = "0.6.15"
```

Comment:
I tried updating this to `0.7` but I get lots of errors like `rust-lld: error: undefined symbol: POWER_CLOCK`

<details>
  <summary>Full output</summary>

```
   Compiling nrf52840-dk-bsp v0.1.0 (/Users/youngrob2/dev/github/robyoung/nrf52840-dk)
error: could not compile `nrf52840-dk-bsp` due to 2 previous errors
[0m[1m[38;5;9merror[0m[0m[1m: linking with `rust-lld` failed: exit status: 1[0m
[0m  [0m[0m[1m[38;5;12m|[0m
[0m  [0m[0m[1m[38;5;12m= [0m[0m[1mnote[0m[0m: "rust-lld" "-flavor" "gnu" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/examples/blinky-33d5c090d7a61b27.1utq84gopygc6c17.rcgu.o" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/examples/blinky-33d5c090d7a61b27.2ww1o1ezi3fs9t8c.rcgu.o" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/examples/blinky-33d5c090d7a61b27.36s3ik9ea1hreqaz.rcgu.o" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/examples/blinky-33d5c090d7a61b27.388ivv4y7kipp0pv.rcgu.o" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/examples/blinky-33d5c090d7a61b27.38jm4v5n63bf4rwu.rcgu.o" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/examples/blinky-33d5c090d7a61b27.3yo4ityqg5m1wcaw.rcgu.o" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/examples/blinky-33d5c090d7a61b27.4yf3uc4lhscmhojz.rcgu.o" "--as-needed" "-L" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps" "-L" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/debug/deps" "-L" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/build/cortex-m-ee6fb7dec2018f51/out" "-L" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/build/cortex-m-rt-fc55061d736c4ea7/out" "-L" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/build/nrf52840-hal-5fd36f571b4583c0/out" "-L" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/build/nrf52840-pac-1dfd25f27c8155c7/out" "-L" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/build/cortex-m-rt-a78069505ccc978a/out" "-L" "/Users/youngrob2/.rustup/toolchains/nightly-aarch64-apple-darwin/lib/rustlib/thumbv7em-none-eabihf/lib" "-Bstatic" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libnrf52840_dk_bsp-eb75d3a8b3068d0e.rlib" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libnrf52840_hal-188aff3ad79955a4.rlib" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libnrf_hal_common-9ce5da635fc1c13f.rlib" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libnrf_usbd-10735e474bded3fa.rlib" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libusb_device-64c1febcdbcb042c.rlib" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libfixed-db10736ef1f7ebc6.rlib" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libbytemuck-fbe13009b4925f56.rlib" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libhalf-38c9464b35b044b9.rlib" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libaz-c8048dfceb48f43d.rlib" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/librand_core-a2433054103e1444.rlib" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libcfg_if-4839a91750e016aa.rlib" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libembedded_storage-b9e2ffa214b32326.rlib" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libembedded_dma-2ee086d21db88dbd.rlib" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libcast-d997c07699182221.rlib" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libnrf52840_pac-9338345780cd56d5.rlib" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libcortex_m_rt-f900bbe96e50247b.rlib" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libr0-bbbe4c4d5511fc3d.rlib" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libcortex_m-5723301b4ba6423c.rlib" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libaligned-4297616563abeb6b.rlib" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libas_slice-610922b8b658a40f.rlib" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libstable_deref_trait-9acd5a6a6274e1d5.rlib" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libgeneric_array-1b55d7a28f404c75.rlib" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libgeneric_array-173757cf8c16e052.rlib" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libgeneric_array-4adef5153736b3a1.rlib" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libtypenum-602fe1f68f5b8080.rlib" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libcortex_m-0cdef69b20585fa3.rlib" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libembedded_hal-fa331011c0674f97.rlib" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libvoid-d04881b1825cd185.rlib" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libnb-4b7b633fb22d46e2.rlib" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libvolatile_register-c4846a4547b7b7de.rlib" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libvcell-878de93acc0125b7.rlib" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libbare_metal-5deba54ecd7009ce.rlib" "--start-group" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libpanic_halt-35ecf81fb93f5758.rlib" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libnb-c667e1b12a47671f.rlib" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libcortex_m_rt-27ac50befa6f1881.rlib" "/Users/youngrob2/.rustup/toolchains/nightly-aarch64-apple-darwin/lib/rustlib/thumbv7em-none-eabihf/lib/librustc_std_workspace_core-9ae03c706dab5b95.rlib" "/Users/youngrob2/.rustup/toolchains/nightly-aarch64-apple-darwin/lib/rustlib/thumbv7em-none-eabihf/lib/libcore-4c81109e506dada5.rlib" "--end-group" "/Users/youngrob2/.rustup/toolchains/nightly-aarch64-apple-darwin/lib/rustlib/thumbv7em-none-eabihf/lib/libcompiler_builtins-2ff80b98fa64b741.rlib" "-Bdynamic" "--eh-frame-hdr" "-znoexecstack" "-L" "/Users/youngrob2/.rustup/toolchains/nightly-aarch64-apple-darwin/lib/rustlib/thumbv7em-none-eabihf/lib" "-o" "/Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/examples/blinky-33d5c090d7a61b27" "--gc-sections" "-Tlink.x"[0m
[0m  [0m[0m[1m[38;5;12m= [0m[0m[1mnote[0m[0m: rust-lld: error: undefined symbol: POWER_CLOCK[0m
[0m          >>> referenced by nrf52840_pac.b04a908e-cgu.10[0m
[0m          >>>               nrf52840_pac-9338345780cd56d5.nrf52840_pac.b04a908e-cgu.10.rcgu.o:(__INTERRUPTS) in archive /Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libnrf52840_pac-9338345780cd56d5.rlib[0m
[0m          [0m
[0m          rust-lld: error: undefined symbol: RADIO[0m
[0m          >>> referenced by nrf52840_pac.b04a908e-cgu.10[0m
[0m          >>>               nrf52840_pac-9338345780cd56d5.nrf52840_pac.b04a908e-cgu.10.rcgu.o:(__INTERRUPTS) in archive /Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libnrf52840_pac-9338345780cd56d5.rlib[0m
[0m          [0m
[0m          rust-lld: error: undefined symbol: UARTE0_UART0[0m
[0m          >>> referenced by nrf52840_pac.b04a908e-cgu.10[0m
[0m          >>>               nrf52840_pac-9338345780cd56d5.nrf52840_pac.b04a908e-cgu.10.rcgu.o:(__INTERRUPTS) in archive /Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libnrf52840_pac-9338345780cd56d5.rlib[0m
[0m          [0m
[0m          rust-lld: error: undefined symbol: SPIM0_SPIS0_TWIM0_TWIS0_SPI0_TWI0[0m
[0m          >>> referenced by nrf52840_pac.b04a908e-cgu.10[0m
[0m          >>>               nrf52840_pac-9338345780cd56d5.nrf52840_pac.b04a908e-cgu.10.rcgu.o:(__INTERRUPTS) in archive /Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libnrf52840_pac-9338345780cd56d5.rlib[0m
[0m          [0m
[0m          rust-lld: error: undefined symbol: SPIM1_SPIS1_TWIM1_TWIS1_SPI1_TWI1[0m
[0m          >>> referenced by nrf52840_pac.b04a908e-cgu.10[0m
[0m          >>>               nrf52840_pac-9338345780cd56d5.nrf52840_pac.b04a908e-cgu.10.rcgu.o:(__INTERRUPTS) in archive /Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libnrf52840_pac-9338345780cd56d5.rlib[0m
[0m          [0m
[0m          rust-lld: error: undefined symbol: NFCT[0m
[0m          >>> referenced by nrf52840_pac.b04a908e-cgu.10[0m
[0m          >>>               nrf52840_pac-9338345780cd56d5.nrf52840_pac.b04a908e-cgu.10.rcgu.o:(__INTERRUPTS) in archive /Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libnrf52840_pac-9338345780cd56d5.rlib[0m
[0m          [0m
[0m          rust-lld: error: undefined symbol: GPIOTE[0m
[0m          >>> referenced by nrf52840_pac.b04a908e-cgu.10[0m
[0m          >>>               nrf52840_pac-9338345780cd56d5.nrf52840_pac.b04a908e-cgu.10.rcgu.o:(__INTERRUPTS) in archive /Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libnrf52840_pac-9338345780cd56d5.rlib[0m
[0m          [0m
[0m          rust-lld: error: undefined symbol: SAADC[0m
[0m          >>> referenced by nrf52840_pac.b04a908e-cgu.10[0m
[0m          >>>               nrf52840_pac-9338345780cd56d5.nrf52840_pac.b04a908e-cgu.10.rcgu.o:(__INTERRUPTS) in archive /Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libnrf52840_pac-9338345780cd56d5.rlib[0m
[0m          [0m
[0m          rust-lld: error: undefined symbol: TIMER0[0m
[0m          >>> referenced by nrf52840_pac.b04a908e-cgu.10[0m
[0m          >>>               nrf52840_pac-9338345780cd56d5.nrf52840_pac.b04a908e-cgu.10.rcgu.o:(__INTERRUPTS) in archive /Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libnrf52840_pac-9338345780cd56d5.rlib[0m
[0m          [0m
[0m          rust-lld: error: undefined symbol: TIMER1[0m
[0m          >>> referenced by nrf52840_pac.b04a908e-cgu.10[0m
[0m          >>>               nrf52840_pac-9338345780cd56d5.nrf52840_pac.b04a908e-cgu.10.rcgu.o:(__INTERRUPTS) in archive /Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libnrf52840_pac-9338345780cd56d5.rlib[0m
[0m          [0m
[0m          rust-lld: error: undefined symbol: TIMER2[0m
[0m          >>> referenced by nrf52840_pac.b04a908e-cgu.10[0m
[0m          >>>               nrf52840_pac-9338345780cd56d5.nrf52840_pac.b04a908e-cgu.10.rcgu.o:(__INTERRUPTS) in archive /Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libnrf52840_pac-9338345780cd56d5.rlib[0m
[0m          [0m
[0m          rust-lld: error: undefined symbol: RTC0[0m
[0m          >>> referenced by nrf52840_pac.b04a908e-cgu.10[0m
[0m          >>>               nrf52840_pac-9338345780cd56d5.nrf52840_pac.b04a908e-cgu.10.rcgu.o:(__INTERRUPTS) in archive /Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libnrf52840_pac-9338345780cd56d5.rlib[0m
[0m          [0m
[0m          rust-lld: error: undefined symbol: TEMP[0m
[0m          >>> referenced by nrf52840_pac.b04a908e-cgu.10[0m
[0m          >>>               nrf52840_pac-9338345780cd56d5.nrf52840_pac.b04a908e-cgu.10.rcgu.o:(__INTERRUPTS) in archive /Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libnrf52840_pac-9338345780cd56d5.rlib[0m
[0m          [0m
[0m          rust-lld: error: undefined symbol: RNG[0m
[0m          >>> referenced by nrf52840_pac.b04a908e-cgu.10[0m
[0m          >>>               nrf52840_pac-9338345780cd56d5.nrf52840_pac.b04a908e-cgu.10.rcgu.o:(__INTERRUPTS) in archive /Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libnrf52840_pac-9338345780cd56d5.rlib[0m
[0m          [0m
[0m          rust-lld: error: undefined symbol: ECB[0m
[0m          >>> referenced by nrf52840_pac.b04a908e-cgu.10[0m
[0m          >>>               nrf52840_pac-9338345780cd56d5.nrf52840_pac.b04a908e-cgu.10.rcgu.o:(__INTERRUPTS) in archive /Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libnrf52840_pac-9338345780cd56d5.rlib[0m
[0m          [0m
[0m          rust-lld: error: undefined symbol: CCM_AAR[0m
[0m          >>> referenced by nrf52840_pac.b04a908e-cgu.10[0m
[0m          >>>               nrf52840_pac-9338345780cd56d5.nrf52840_pac.b04a908e-cgu.10.rcgu.o:(__INTERRUPTS) in archive /Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libnrf52840_pac-9338345780cd56d5.rlib[0m
[0m          [0m
[0m          rust-lld: error: undefined symbol: WDT[0m
[0m          >>> referenced by nrf52840_pac.b04a908e-cgu.10[0m
[0m          >>>               nrf52840_pac-9338345780cd56d5.nrf52840_pac.b04a908e-cgu.10.rcgu.o:(__INTERRUPTS) in archive /Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libnrf52840_pac-9338345780cd56d5.rlib[0m
[0m          [0m
[0m          rust-lld: error: undefined symbol: RTC1[0m
[0m          >>> referenced by nrf52840_pac.b04a908e-cgu.10[0m
[0m          >>>               nrf52840_pac-9338345780cd56d5.nrf52840_pac.b04a908e-cgu.10.rcgu.o:(__INTERRUPTS) in archive /Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libnrf52840_pac-9338345780cd56d5.rlib[0m
[0m          [0m
[0m          rust-lld: error: undefined symbol: QDEC[0m
[0m          >>> referenced by nrf52840_pac.b04a908e-cgu.10[0m
[0m          >>>               nrf52840_pac-9338345780cd56d5.nrf52840_pac.b04a908e-cgu.10.rcgu.o:(__INTERRUPTS) in archive /Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libnrf52840_pac-9338345780cd56d5.rlib[0m
[0m          [0m
[0m          rust-lld: error: undefined symbol: COMP_LPCOMP[0m
[0m          >>> referenced by nrf52840_pac.b04a908e-cgu.10[0m
[0m          >>>               nrf52840_pac-9338345780cd56d5.nrf52840_pac.b04a908e-cgu.10.rcgu.o:(__INTERRUPTS) in archive /Users/youngrob2/dev/github/robyoung/nrf52840-dk/target/thumbv7em-none-eabihf/debug/deps/libnrf52840_pac-9338345780cd56d5.rlib[0m
[0m          [0m
[0m          rust-lld: error: too many errors emitted, stopping now (use -error-limit=0 to see all errors)[0m
[0m          [0m

[0m[1m[38;5;9merror[0m[0m[1m: aborting due to previous error[0m

       Error Failed to build the cargo project.

        Hint 'cargo build' was not successful. Have a look at the error output above.

        Hint Make sure the working directory you selected is indeed a cargo project with a Cargo.toml in it.
```
</details>

Reply:
I've seen these kinds of errors recently. The reason were multiple cortex-m-rt versions being included in the build. I suppose this caused the `device.x` from the PAC to not be included.

I think you can't update to a `cortex-m-rt` to one that doesn't match the underlying dependencies, but if this crate isn't affected by the breaking changes, you can specifiy `cortex-m-rt = ">=0.6, <0.8"` (or something along those lines) to ease the transition.

Reply:
Ahh, I see now, cargo tree is showing me that the `nrf52840-pac` crate includes `cortex-m-rt v0.6.15`. Good idea on the looser dependency. Done.